CONVENTIONS.md

```diff
@@ -0,0 +1,75 @@
+ - Prefer clarification in the documentation comment over verbose names. Conversely, do not give fields,
+   methods, or method parameters useless comments.
+
+General
+-------
+ - All names use US spelling.
+ - Acronyms should generally be written all-lowercase, while normal camel case rules still apply.
+   For instance, "identifier" becomes `id` as standalone word or at the beginning of a camel case phrase and
+   `Id` when used within a phrase.<br>
+   There are various exceptions to this rule which apply only when the word does not start with them:
+     - axis-aligned bounding boxes: `AxisAlignedBB` → `AABB`
+     - named binary tags: `NBT`
+     - red, green, blue (and alpha) color components → `RGB(A)`
+
+Type Names
+----------
+ - New classes, interfaces etc. should follow a suffix-based naming scheme.
+   Specifically, this means for any name, `NameType` would be the correct name, where `Type` should be
+   derived from the supertypes.
+ - Abstract base classes should be prefixed with `Abstract`.
+ - Interfaces are prefixed with the letter `I`.
+ - Enums should be named directly after the things they are enumerating, in singular form.
+   There is no `Enum` prefix/suffix.
+
+**Note:** Long-established names where prefixes are used (subclasses `Block`, `Item` etc.)
+do *not* fall under the suffix convention.
+
+### Examples
+ - For a new shelf block with an inventory, appropriate names would be `BlockShelf`, `TileEntityShelf` and
+   `ShelfContainer`.
+
+Field Names
+-----------
+ - `boolean` type fields should *not* start with an *is* prefix unless there is a convincing reason to do so.
+ - No prefixes like `the` or suffixes like `Obj`.
+
+Method Names
+------------
+ - Method names should always start with a verb. Standard conventions for setters and getters apply.
+ - Methods returning a `boolean` should have a name beginning with an appropriate prefix from the following list:
+     - `can` 
+     - `does`
+     - `has`
+     - `is`
+     - `should`
+     - `was`
+ - 'Event handling' methods should follow an `on<Noun>` theme where the noun is the event's name
+   (with redundancies removed).
+ - Methods which read or write NBT should be prefixed `read` and `write` respectively.
+ - Methods which deseralize or serialize JSON should have a `deserialize` or `serialize` prefix accordingly. 
+
+### Examples
+ - The method for handling block right clicks is called `onActivation` (noun form, no redundancy),
+   not `onBlockActivated` (past participle, redundant `Block` infix).
+
+Method Parameter Names
+----------------------
+ - If a name clashes with that of a field or a type name, it should be suffixed with `In`
+ - Parameters of certain types should always get named according to the following list:
+     - `BlockPos` arguments should be named `pos` when there is a single one, and `*Pos` when there are multiple
+     - Parameters of the `IWorldReaderBase` and related type should be `worldIn`
```

Comment:
I think it's important to qualify this with a "where the behavior is used simply as the world to perform the operation on" (and a similar thing for `BlockPos`).  For example, `p_72375_2_` for `PlayerList.preparePlayer` used to be named `worldIn`, but it's actually the world the player was in _previously_ and has been renamed to `oldWorld`.  Basically, look at it first and make sure the usage is conventional; this isn't an "always" rule.

Reply:
Agreed, this shouldn't be an always rule, and there's cases where the world is not a world, but a `WorldGenRegion`, which I'd say is more applicable to `region`, but that's an argument outside the scope of this PR.

Reply:
I've added this as further clarification.